tests/ga/test_update.py

```diff
@@ -1341,15 +1341,6 @@ def _test_upgrade_available(
     def test_upgrade_available_returns_true_on_first_use(self):
         self.assertTrue(self._test_upgrade_available())
 
-    def test_upgrade_available_will_refresh_goal_state(self):
```

Comment:
why did we remove this test altogether? (instead of updating it?)

Reply:
The behaviour of the function we're testing changed; the retry on `ResourceGoneError` and forceful refresh of the goal state is now done elsewhere. I've added tests for the new behaviour, which is why I removed this test.